scheduling/src/uppaal_executor.cpp

```diff
@@ -59,42 +61,96 @@ std::string scheduling::UppaalExecutor::execute()
             throw SchedulingException("Could not start verifyta. errno: " + std::to_string(errno) +
                                       ".");
         }
-
-        return "";
     }
     else {
+        reset_pid(pid);
         // Parent
         close(fd[CHILD_WRITE]);
         close(fd[CHILD_READ]);
+        close(fd[PARENT_WRITE]);
 
         // Wait for completion
-        std::cout << "Waiting for completion...\n";
-        int status;
-        waitpid(pid, &status, NO_FLAGS);
-        std::cout << "Scheduling complete with status " << status << ".\n";
+        std::cout << "Waiting for completion..." << std::endl;
+        worker = std::thread([&, parent_read = fd[PARENT_READ], callback]() -> void {
+            int status;
+            int pid;
+
+            std::unique_lock get_pid{pid_lock};
+            if (!child_pid.has_value()) {
+                return; // TODO better behaviour?
```

Comment:
Er det en TODO for et andet pull request?

Reply:
Nææ det er bare en overvejelse.